pkg/evaluation/util.go

```diff
@@ -0,0 +1,90 @@
+package evaluation
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/harness/ff-golang-server-sdk/log"
+	"github.com/harness/ff-golang-server-sdk/rest"
+	"github.com/spaolacci/murmur3"
+)
+
+func getAttrValue(target *rest.Target, attr string) reflect.Value {
+	var value reflect.Value
+	attrs := make(map[string]interface{})
+	if target.Attributes != nil {
+		attrs = *target.Attributes
+	}
+
+	attrVal, ok := attrs[attr] // first check custom attributes
+	if ok {
+		value = reflect.ValueOf(attrVal)
+	} else {
+		// We only have two fields here, so we will access the fields directly, and use reflection if we start adding
+		// more in the future
+		switch strings.ToLower(attr) {
+		case "identifier":
+			value = reflect.ValueOf(target.Identifier)
+		case "name":
+			value = reflect.ValueOf(target.Name)
+		}
+	}
+	return value
+}
+
+func findVariation(variations []rest.Variation, identifier string) (rest.Variation, error) {
+	for _, variation := range variations {
+		if variation.Identifier == identifier {
+			return variation, nil
+		}
+	}
+	return rest.Variation{}, fmt.Errorf("%w: %s", ErrVariationNotFound, identifier)
+}
+
+func getNormalizedNumber(identifier, bucketBy string) int {
+	value := []byte(strings.Join([]string{bucketBy, identifier}, ":"))
+	hasher := murmur3.New32()
+	_, err := hasher.Write(value)
+	if err != nil {
+		log.Debugf("error %v", err)
+	}
+	hash := int(hasher.Sum32())
+	return (hash % oneHundred) + 1
+}
+
+func isEnabled(target *rest.Target, bucketBy string, percentage int) bool {
+	value := getAttrValue(target, bucketBy)
+	identifier := value.String()
+	if identifier == "" {
+		return false
+	}
+
+	bucketID := getNormalizedNumber(identifier, bucketBy)
+	return percentage > 0 && bucketID <= percentage
+}
+
+func evaluateDistribution(distribution *rest.Distribution, target *rest.Target) string {
+	variation := ""
+	if distribution == nil {
+		return variation
+	}
+
+	for _, wv := range distribution.Variations {
+		variation = wv.Variation
+		if isEnabled(target, distribution.BucketBy, wv.Weight) {
```

Comment:
I know this is existing code, but I think there might be a bug in this, if there is more than two weights.  i.e. lets say we have three variations offer1, offer2, offer3

offer1 = 10%
offer2 = 20%
offer3 = 70%

I think that Weight needs to be incremented following each call, otherwise users will fall into the same buckets.  i.e. lets say offer1 is the first 10% you want users who fall between 0-10 out of 100.  Offer2 as it is will look at users between 0-20 out of 100 where I think we want it to look for users that get put into a bucket for the range 10-30 out of 100.   

I'm not certian about this, I've created a playground sample, so maybe you can take a look and confirm/verify ?
https://go.dev/play/p/KdnIczDj7R1